src/openms/include/OpenMS/ANALYSIS/ID/SiriusAdapterAlgorithm.h

```diff
@@ -54,19 +56,84 @@ namespace OpenMS
       SiriusAdapterAlgorithm();
 
       /// Struct for temporary folder structure
-      struct SiriusTmpStruct
+      struct SiriusTemporaryFileSystemObjects
       {
-        String tmp_dir;
-        String tmp_ms_file;
-        String tmp_out_dir;
-      };
+      public:
 
-      /**
-        @brief Construct temporary folder structure for SIRIUS (SiriusTmpStruct)
+        SiriusTemporaryFileSystemObjects(int debug_level) : removeable_(false) { constructSiriusTemporaryFileSystemObjects_(debug_level); }
+
+        ~SiriusTemporaryFileSystemObjects(){ if (removeable_) removeSiriusTemporaryFileSystemObjects_(debug_level_, tmp_dir_, tmp_ms_file_); }
+
+        String getTmpDir()
```

Comment:
should be `const String& getTmpDir() const;`

Reply:
similarly for the other accessors

Reply:
Just for my understanding: The second const (after the function) will throw a compiler error for this class function if a member variable of the class will be changed. 

So it is basically a safety mechanism, or does it have further additional functionality?